sw/device/lib/dif/dif_spi_device.h

```diff
@@ -0,0 +1,243 @@
+    dif_spi_device_t *spi);
+
+/**
+ * Issues an "abort" to the given SPI device, causing all in-progress IO to
+ * halt.
+ *
+ * @param spi A SPI device.
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_abort(const dif_spi_device_t *spi);
+
+/**
+ * Resets all interrupt-related state on the given SPI device, such as enabled
+ * interrupts and set RX/TX levels.
+ *
+ * @param spi A SPI device.
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_irq_reset(const dif_spi_device_t *spi);
+
+/**
+ * Checks whether a particular interrupt bit has been set.
+ *
+ * @param spi A SPI device.
+ * @param type Which IRQ type to check.
+ * @param flag Out param for the interrupt bit.
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_irq_get(const dif_spi_device_t *spi,
+                                               dif_spi_device_irq_type_t type,
+                                               bool *flag_out);
+
+/**
+ * Clears all active interrupt bits.
+ *
+ * @param spi A SPI device.
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_irq_clear_all(
+    const dif_spi_device_t *spi);
+
+/**
+ * Enable a particular interrupt enable bit to enabled or disabled.
+ *
+ * @param spi A SPI device.
+ * @param type Which IRQ type to toggle.
+ * @param state The state to update the bit to.
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_irq_enable(
+    const dif_spi_device_t *spi, dif_spi_device_irq_type_t type,
+    dif_spi_device_irq_state_t state);
+
+/**
+ * Forces a particular IRQ type to fire.
+ *
+ * @param spi A SPI device.
+ * @param type Which IRQ type to fire.
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_irq_force(
+    const dif_spi_device_t *spi, dif_spi_device_irq_type_t type);
+
+/**
+ * Sets up the "FIFO level" (that is, number of bytes present in a particular
+ * FIFO) at which the TxLevel and RxLevel IRQs will fire.
+ *
+ * For the reciept side, when the FIFO overflows `rx_level`, an interrupt is
+ * fired. This can be used to detect that more data should be read from the
+ * RX FIFO. This is the `SRAM -> Main Memory` case.
+ *
+ * Conversely, for the transmission side, when the FIFO underflows `tx_level`,
+ * an interrupt is fired. This can be used to detect that there is free space to
+ * write more data to the TX FIFO. This is the `Main Memory -> SRAM` case.
+ *
+ * @param spi A SPI device.
+ * @param rx_level The new RX level, as described above.
+ * @param tx_level The new TX level, as described above.
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_set_irq_levels(
+    const dif_spi_device_t *spi, uint16_t rx_level, uint16_t tx_level);
+
+/**
+ * Checks how many bytes are still pending reciept by software in the RX FIFO.
```

Comment:
`receipt`. Also I don't like `Checks`. If it's checking then what's it doing? I would say "Returns the number of bytes..."

Reply:
Done.